### PR TITLE
Save to temp to prevent file corruption

### PR DESCRIPTION
**Pull request by r888800009**

### CODE REVIEW

**Comment by r888800009** [comment]
Several JSON issues may be caused by a corrupted configuration file.
https://github.com/Inochi2D/inochi-creator/issues/167
https://github.com/Inochi2D/inochi-creator/issues/387
https://github.com/Inochi2D/inochi-creator/issues/332

**Comment by seagetch** [comment]
@r888800009 
For serialization of model to file, inWriteINPPuppet serialize contents to memory first, and then dump it to file.
So if file is corrupted, that means memory content is already corrupted.

I guess copying from swp to inp cannot be much help. (because it already write broken value to the memory successfully.)
In my experience, some times JSON output n/a values, and failed on loading. Checking that kind of buggy output or safe guard in loading is important.
Root problem is that "n/a" is handled as string in JSON, but loader expect it to be read as floating value.

Catching any exceptions during inWriteINPPuppetMemory would be beneficial to prevent broken content to file. But I think preventing n/a value is most important.

**Comment by r888800009** [comment]
@seagetch 
The `write()` function on POSIX systems is not atomic unless the write size is smaller than `PIPE_BUF`, and most `.inp` use cases are likely to exceed this size.

Therefore, it is important to ensure atomicity guarantees to prevent unexpected issues in the event of a system crash or power loss.

**Comment by r888800009** [comment]
I haven’t investigated `nan` specifically yet. I have a test file with `nan` in `uvs`. I'm not sure if other fields also have issues, but should the standard library be able to handle this? It might be a problem with parameters.

**Comment by seagetch** [comment]
usually deformations have highest possibility to have nan.

**Comment by seagetch** [comment]
deformation is updated every time after editing vertices. there seems edge case that this calculation generate nan value.

File: source/nijigenerate/core/settings.d [hunk 1]
```diff
@@ -42,7 +42,10 @@ void incSettingsLoad() {
     Saves settings from settings store
 */
 void incSettingsSave() {
-    write(incSettingsPath(), settings.toString());
+    // using swp prevent file corruption
+    string swapPath = incSettingsPath() ~ ".swp";
+    write(swapPath, settings.toString());
+    rename(swapPath, incSettingsPath());
```
[the remark this answers is not in the document]
Comment (r888800009) [reply source/nijigenerate/core/settings.d:48]: I think most programming languages ​​compiled into IR or machine code sequences will first execute `settings.toString()`, and a crash will occur in this function `toString()`, will not execute `write()`

[the remark this answers is not in the document]
Comment (r888800009) [reply source/nijigenerate/core/settings.d:48]: For example, this code is as follows
```d
string toString() {return "hi";}
void write(string path, string data) {
    // do something
}
int main() {
    write("/tmp/tmp", toString());
    return 0;
}
```
using godbolt and ldc
Assembly language should be compiled into 
```asm
_Dmain:
        push    rbp
        mov     rbp, rsp
        sub     rsp, 16
        call    immutable(char)[] example.toString()@PLT
        mov     qword ptr [rbp - 8], rax
        mov     rcx, rdx
        mov     rdx, qword ptr [rbp - 8]
        lea     rsi, [rip + .L.str.1]
        mov     edi, 8
        call    void example.write(immutable(char)[], immutable(char)[])@PLT
        xor     eax, eax
        add     rsp, 16
        pop     rbp
        ret
```
So we can confirm that toString will be called first

**Pull request merged**